### PR TITLE
Add an .editorconfig.

### PR DESCRIPTION
**Pull request by jpobst**
There may be a few tweaks to make later, but this should get us very close to what our style is.

The main ambiguities are:

### Namespace and Class open braces on a new line

Mono Coding Guidelines say this:
```
namespace Blah {
        class Foo {
        }
}
```

but the vast majority of XA code says this:
```
namespace Blah
{
        class Foo
        {
        }
}
```

I went with reality (our code) rather than the Mono guidelines.

### Spaces before square brackets

There is only one setting for spaces before square brackets.  Our code seems to generally use no space for declaration and a space for invocation:
```
int[] counts;
Console.WriteLine (counts [0]);
```

The Mono Guidelines only specify a space before invocation and don't specify anything about declaration, so I went with a space, thus:
```
int [] counts;
Console.WriteLine (counts [0]);
```

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+trim_trailing_whitespace = true
+max_line_length = 180
+
+# MSBuild
+[*.{csproj,proj,projitems,shproj,fsproj,target,props}]
+indent_style = space
+indent_size = 2
+
+# XML config files
+[*.{config,nuspec,resx}]
```
Comment (jonpryor) [inline .editorconfig]: This should also list `xml`, `axml`.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+# All files
+[*]
+charset = utf-8
+indent_style = tab
+tab_width = 8
+trim_trailing_whitespace = true
+max_line_length = 180
+
+# MSBuild
+[*.{csproj,proj,projitems,shproj,fsproj,target,props}]
```
Comment (jonpryor) [inline .editorconfig]: It's `targets` (plural), not `target` (singular).

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+csharp_new_line_before_catch = false
+csharp_new_line_before_finally = false
+csharp_new_line_before_members_in_object_initializers = true
+csharp_new_line_before_members_in_anonymous_types = true
+
+# Indentation preferences
+csharp_indent_switch_labels = false
+
+# Space preferences
+csharp_space_after_cast = false
```
Comment (jonpryor) [inline .editorconfig]: I think this should be `true`.

**Comment by jonpryor** [comment]
This should also include anything relevant from PR #2843, which appears to just be:

```
csharp_indent_case_contents = true
csharp_indent_switch_labels = true	
```

**Comment by jonpryor** [comment]
For eventual commit message purposes: 

* https://editorconfig.org
* https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference

**Comment by jpobst** [comment]
Good feedback, updated!

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+trim_trailing_whitespace = true
+max_line_length = 180
+
+# MSBuild
+[*.{csproj,proj,projitems,shproj,fsproj,targets,props}]
+indent_style = space
+indent_size = 2
+
+# XML config files
+[*.{xml,axml,config,nuspec,resx}]
```
Comment (jonathanpeppers) [inline .editorconfig]: `xaml` too? We have a Xamarin.Forms test app.

Comment (jpobst) [reply .editorconfig]: Added!

**Pull request merged**